src/currency-field/CurrencyField.ts

```diff
@@ -105,6 +105,27 @@ export class CurrencyField extends OmniFormElement {
         }
     }
 
+    override async attributeChangedCallback(name: string, _old: string | null, value: string | null): Promise<void> {
+        super.attributeChangedCallback(name, _old, value);
+        if (name === 'value') {
+            if (typeof value === 'string') {
+                await this._formatToCurrency(value).then((res) => {
+                    this._inputElement!.value = res;
```

Comment:
Should we not rather have null checks here? Sometimes attributeChanged can run early and the element might not be available yet.

Reply:
Noted, implementing the suggested change.